pulp_rpm/app/serializers.py

```diff
@@ -208,13 +217,31 @@ def validate(self, data):
 
         """
         data = super().validate(data)
+        # export META from rpm and prepare dict as saveable format
+        try:
+            new_pkg = _prepare_package(data["artifact"], data["relative_path"])
+        except OSError:
+            raise OSError('RPM file cannot be parsed for metadata.')
+
+        package = Package.objects.filter(**new_pkg)
+
+        if package.exists():
+            keywords = ('name', 'epoch', 'version', 'release', 'arch', 'checksum_type', 'pkgId')
+            error_data = ", ".join(
+                ["=".join(item) for item in new_pkg.items() if item[0] in keywords]
+            )
 
-        data['relative_path'] = data.pop('relative_path')
+            raise serializers.ValidationError(
+                _(
+                    "There is already a package with: {values}."
+                ).format(values=error_data)
+            )
 
+        data.update(new_pkg)
```

Comment:
given: https://pulp.plan.io/issues/5403?pn=1#note-9
This method should not overwrite `validate`, but `deferred_validate` which will only be called in the triggered task.

Reply:
when I change some tests break due to read_only fields:
https://travis-ci.org/pulp/pulp_rpm/jobs/587603270#L1363